src/gleam/http.gleam

```diff
@@ -368,18 +368,30 @@ pub fn default_req() -> Request(String) {
     body: "",
     scheme: Https,
     host: "localhost",
-    port: None,
+    port: option.None,
     path: "",
-    query: None,
+    query: option.None,
   )
 }
 
+/// Set the scheme (protocol) of the request.
+///
+pub fn set_scheme(req: Request(body), scheme: Scheme) -> Request(body) {
+  Request(..req, scheme: scheme)
+}
+
 /// Set the method of the request.
 ///
 pub fn set_host(req: Request(body), host: String) -> Request(body) {
   Request(..req, host: host)
 }
 
+/// Set the port of the request.
+///
+pub fn set_port(req: Request(body), port: Option(Int)) -> Request(body) {
```

Comment:
Can we make this take just an Int for the port? Can wrap with `Sone` in the function body.

Reply:
Based on my experience with pipeline builders like I think it is more flexible to require the optional type. 

Consider the case when you get the port via an argument from a different place, you will probably have an Option.
Then you can pass that option directly to `set_port(maybe_port)`.

But `set_port` if only takes an `Int`, then you have to call that conditionally. Call set_port if your option is Some, don't call it if None.

Also if `set_port` only takes an Int, maybe we should have `unset_port` as well. 

Another solution would be to have `set_port` take an `Int` and have a related function that takes the `Option`. e.g. `maybe_set_port`.

So, my preference for this is to take the `Option`

Reply:
My thoughts on this is I can't think of any situation in which someone would want to remove the port from a request. I could see reason to overwrite an existing port, but not remove it in this fashion. Because of this I think there is value in in having an API that is more ergonomic for the use that is the most common. The record update syntax can be used in other scenarios.

Reply:
My concern is not about removing the port. Is about getting the port input from a different source, and having that input be an `Option`. Then you have to do something like:

```
fn send_request(maybePort: Option(Int), ...) {
  fn maybe_set_port(req) {
    case maybePort {
       Some(port) => http.set_port(port)
       None => function.identity
    }
  }

  req
  |> maybe_set_port
  |> ...
}
```

Would you be open to adding two functions for this? 
`set_port(Int)` and `set_optional_port(Option(Int))`

Most of the time you would use `set_port(Int)`. But then we have `set_optional_port` fot this special case.

Reply:
Unless there's particular use case that would pass though an option like that I would like to hold off. We can add it later once someone has need for it. It is also easy for the user to implement as you've shown, so it won't block anyone in their work.